MVC-App/Services/RelationService.cs

```diff
@@ -200,36 +193,40 @@ public async Task<RelationListVM> GetAsync(Guid? categoryId)
 
         public async Task Create(CreateEditRelationVM relationVM)
         {
-            var relation = new Relation
-            {
-                Id = Guid.NewGuid(),
-                Name = relationVM.Relation.Name,
-                FullName = relationVM.Relation.FullName,
-                TelephoneNumber = relationVM.Relation.TelephoneNumber,
-                EMailAddress = relationVM.Relation.Email,
-                CreatedAt = DateTime.Now,
-                CreatedBy = "admin",
-                IsDisabled = false,
-                IsTemporary = false,
-                IsMe = false,
-                PaymentViaAutomaticDebit = false,
-                InvoiceDateGenerationOptions = 0,
-                InvoiceGroupByOptions = 0
-            };
+            var config = new MapperConfiguration(cfg => cfg.CreateMap<RelationVM, Relation>()
+            .AfterMap((src, dest) => {
+                dest.Id = Guid.NewGuid();
+                dest.EMailAddress = src.Email;
+                dest.CreatedAt = DateTime.Now;
+                dest.CreatedBy = "admin";
+                dest.IsDisabled = false;
+                dest.IsTemporary = false;
+                dest.IsMe = false;
+                dest.PaymentViaAutomaticDebit = false;
+                dest.InvoiceDateGenerationOptions = 0;
+                dest.InvoiceGroupByOptions = 0;
+            }
+            ));
+
+            var mapper = config.CreateMapper();
+
+            var relation = mapper.Map<Relation>(relationVM.Relation);
 
             _unitOfWork.RelationRepository.Insert(relation);
 
-            var relationAddress = new RelationAddress
-            {
-                Id = Guid.NewGuid(),
-                RelationId = relation.Id,
-                CountryId = relationVM.Relation.CountryId,
-                City = relationVM.Relation.City,
-                Street = relationVM.Relation.Street,
-                PostalCode = ApplyMask(relationVM.Relation.PostalCode, _unitOfWork.CountryRepository.GetByID(relationVM.Relation.CountryId).PostalCodeFormat),
-                Number = relationVM.Relation.StreetNumber,
-                AddressTypeId = Guid.Parse("00000000-0000-0000-0000-000000000002")
-            };
+            config = new MapperConfiguration(cfg => cfg.CreateMap<RelationVM, RelationAddress>()
+            .AfterMap((src, dest) => {
+                dest.Id = Guid.NewGuid();
+                dest.RelationId = relation.Id;
+                dest.PostalCode = ApplyMask(src.PostalCode, _unitOfWork.CountryRepository.GetByID(relationVM.Relation.CountryId).PostalCodeFormat);
+                dest.Number = src.StreetNumber;
```

Comment:
Remove
dest.Number = src.StreetNumber;
And test if it works.

Reply:
same issue, mapper fails to map exact field